processing/src/main/java/io/druid/query/Queries.java

```diff
@@ -55,6 +61,18 @@ public static void verifyAggregations(
             "Missing fields [%s] for postAggregator [%s]", missing, postAgg.getName()
         );
         Preconditions.checkArgument(combinedAggNames.add(postAgg.getName()), "[%s] already defined", postAgg.getName());
+
+        if (postAgg instanceof HasDependentAggFactories) {
```

Comment:
A Ut for a finalizing post aggregator with arithmetic post aggregator has beed added. It seems work, but do you mean this case or other situation? Of course, I agree that we can use better ways to get aggregators injected. Would you give me some recommandation for this?